src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,137 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private static final int INITIAL_SIZE = 0;
+    private static final int INITIAL_INDEX = 0;
+    private static final String TEXT_OF_EXCEPTION = "Invalid index outside"
+            + " the dimension of the array";
+    private int size;
+    private T[] elementData;
```

Comment:
let's rename in plural manner to be more clear through the code that this is not one element.
```suggestion
    private T[] elementsData;
```